editoast/src/views/projects.rs

```diff
@@ -301,102 +302,166 @@ impl From<ProjectPatchForm> for Changeset<Project> {
 async fn patch(
     data: Json<ProjectPatchForm>,
     project_id: Path<i64>,
-    db_pool: Data<DbConnectionPool>,
+    db_pool: Data<DbConnectionPoolV2>,
 ) -> Result<Json<ProjectWithStudyCount>> {
+    let conn = &mut db_pool.get().await?;
     let data = data.into_inner();
     let project_id = project_id.into_inner();
     if let Some(image) = data.image {
-        check_image_content(db_pool.clone(), image).await?;
+        check_image_content(conn, image).await?;
     }
     let project_changeset: Changeset<Project> = data.into();
-    let conn = &mut db_pool.get().await?;
     let project = Project::update_and_prune_document(conn, project_changeset, project_id).await?;
     Ok(Json(ProjectWithStudyCount::try_fetch(conn, project).await?))
 }
 
 #[cfg(test)]
 pub mod test {
-    use actix_http::Request;
+    use std::ops::DerefMut;
+
     use actix_web::http::StatusCode;
-    use actix_web::test as actix_test;
+    use actix_web::test::call_and_read_body_json;
     use actix_web::test::call_service;
-    use actix_web::test::read_body_json;
     use actix_web::test::TestRequest;
     use rstest::rstest;
     use serde_json::json;
-    use std::sync::Arc;
 
     use super::*;
-    use crate::fixtures::tests::db_pool;
-    use crate::fixtures::tests::project;
-    use crate::fixtures::tests::TestFixture;
-    use crate::modelsv2::DeleteStatic;
-    use crate::views::tests::create_test_service;
-
-    fn delete_project_request(project_id: i64) -> Request {
-        TestRequest::delete()
-            .uri(format!("/projects/{project_id}").as_str())
-            .to_request()
+    use crate::modelsv2::prelude::*;
+    use crate::views::test_app::TestAppBuilder;
+
+    pub fn project_changeset() -> Changeset<Project> {
+        Project::changeset()
+            .name("test_project".to_owned())
+            .budget(Some(0))
+            .creation_date(Utc::now().naive_utc())
+            .last_modification(Utc::now().naive_utc())
+            .tags(Tags::default())
+    }
+
+    async fn create_project(pool: &DbConnectionPoolV2) -> Project {
+        project_changeset()
+            .create(pool.get_ok().deref_mut())
+            .await
+            .expect("Failed to create project")
```

Comment:
This is mostly duplication of the functions in `modelsv2::project::test`. You could define a `modelsv2::fixtures` public module that could be reused in the views.